docs/guide/extending/extending_thumbnails.md

```diff
@@ -0,0 +1,128 @@
+# Extending thumbnails
+
+Thumbnails enable you to easily choose an image for a specific Content Type.
+If you do not want to use custom thumbnails [`ContentType` icons](../../resources/icons.md#content-types) will be used instead.
+
+## Thumbnail mechanism 
+
+The thumbnail mechanism has two layers, and each layer can have many implementations.
```

Comment:
I would start by describing the default process, the one that uses the "Use as thumbnail" checkbox in fields definitions, as it is what most users will face on a daily basis. Then I would go into details.